core/test/unit/middleware/check-ssl_spec.js

```diff
@@ -41,6 +41,29 @@ describe('checkSSL', function () {
         done();
     });
 
+    it('should redirect to subdirectory (frontend)', function (done) {
+        req.originalUrl = '/blog';
+        // req.url can be different than req.originalUrl when mounting or url
+        // rewriting happens.
+        req.url = '/';
+        res.isAdmin = true;
```

Comment:
Your tests are now red, because the middleware does not support this use case.
And i **think** this is correct. We only redirect the admin requests to HTTPS. Can you help? @ErisDS